src/htm/algorithms/SpatialPooler.cpp

```diff
@@ -429,12 +429,7 @@ void SpatialPooler::initialize(
   iterationNum_ = 0u;
   iterationLearnNum_ = 0u;
 
-  tieBreaker_.resize(numColumns_);
-  for (Size i = 0; i < numColumns_; i++) {
-    tieBreaker_[i] = (Real)(0.01 * rng_.getReal64());
-  }
-
-  overlapDutyCycles_.assign(numColumns_, 0);
+  overlapDutyCycles_.assign(numColumns_, 0); //TODO make all these sparse or rm to reduce footprint
```

Comment:
I think we can get rid of `overlapDutyCycles_` altogether.  IIRC it is a boosting method?  We have better/stronger boosting methods.  I don't think that this even ever kicks in?  But that needs to be checked before removing it, an issue for a different PR

Reply:
I think it's used for something...bumpupWeakSegments or so.. I have waiting another PR that puts all of boosting code into 2 methods: one for boosting overlaps, other for weak columns (and shows the "weak columns" is not really useful, on the limited mnist, hotgym tests) But yes, another PR